src/Abstractions/CharBoard.cs

```diff
@@ -72,4 +76,142 @@ public CharBoard Set(int r, int c, char value)
             return new CharBoard(Size, newCells);
         }
     }
+    
+    public record DiceBoard
+    {
+        private static readonly ConcurrentDictionary<int, DiceBoard> EmptyCache = new();
+        public static DiceBoard Empty(int size) => EmptyCache.GetOrAdd(size, size1 => new DiceBoard(size1));
+
+        public int Size { get; }
+        public ImmutableDictionary<int, Cell> Cells { get; }
+        
+        public struct Cell
```

Comment:
This thing definitely requires some serialization-related optimizations. Or maybe DiceBoard.